Share/NCShareExtension.swift

```diff
@@ -177,17 +177,22 @@ class NCShareExtension: UIViewController {
         collectionView.reloadData()
         tableView.reloadData()
     }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        self.cancel(with: .cancel)
+    }
 
     // MARK: -
 
     func cancel(with error: NCShareExtensionError) {
         // make sure no uploads are continued
         uploadStarted = false
-        let metadata = uploadMetadata[counterUploaded]
-        let filePath = CCUtility.getDirectoryProviderStorageOcId(metadata.ocId, fileNameView: metadata.fileNameView)!
-
-        NCManageDatabase.shared.deleteMetadata(predicate: NSPredicate(format: "ocId == %@", metadata.ocId))
-        NCNetworking.shared.uploadRequest[filePath]?.tasks.forEach({ $0.cancel() })
+        for metadata in uploadMetadata {
+            let filePath = CCUtility.getDirectoryProviderStorageOcId(metadata.ocId, fileNameView: metadata.fileNameView)!
+            NCManageDatabase.shared.deleteMetadata(predicate: NSPredicate(format: "ocId == %@", metadata.ocId))
+            NCNetworking.shared.uploadRequest[filePath]?.tasks.forEach({ $0.cancel() })
+        }
         extensionContext?.cancelRequest(withError: error)
```

Comment:
if the user has remove the last or one file and close the share the app crashed without call the extensionContext?.cancelRequest

Reply:
I mean the user cannot remove metadata while it's being uploaded but yea.. makes sense 👍